File: 20221213/Carpet.test.js
import solution from './Carpet';

test('solution', () => {
  expect(solution(10, 2)).toBe([4, 3]);
  expect(solution(8, 1)).toBe([3, 3]);
  expect(solution(24, 24)).toBe([8, 6]);
});

test('calsNeededYellow', () => {
  expect(solution(10, 1)).toBe(2);
  expect(solution(8, 1)).toBe(1);
  expect(solution(24, 4)).toBe(24);
});

File: 2023/20230329/Carpet/src/test/java/CarpetTest.java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CarpetTest {
    private Carpet test;

    @BeforeEach
    void setup() {
        test = new Carpet();
    }

    @Test
    void solution() {
        assertArrayEquals(new int[] {4, 3}, test.solution(10, 2));
        assertArrayEquals(new int[] {3, 3}, test.solution(8, 1));
        assertArrayEquals(new int[] {8, 6}, test.solution(24, 24));
    }

    @Test
    void calcNecessaryYellow() {
        assertEquals(2, test.calcNecessaryYellow(10, 1));
    }
}
